i6_models/parts/conformer/convolution.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+import torch
+from torch import nn
+
+
+class ConformerConvolutionV1(nn.Module):
+    """
+    Conformer convolution module.
+    see also: https://github.com/espnet/espnet/blob/713e784c0815ebba2053131307db5f00af5159ea/espnet/nets/pytorch_backend/conformer/convolution.py#L13
+    """
+
+    def __init__(self, channels: int, kernel_size: int, dropout: float = 0.1, activation: nn.Module = nn.SiLU()):
```

Comment:
Actually, `dropout` should be completely removed here, right? And then you apply dropout in the outer module, just like applying the residual connection.

Reply:
Why would you prefer that?

Reply:
Because that makes it more flexible and cleaner. E.g. in Transformer, it is common that people play around with the order of layer-norm, dropout and residual connection. I have seen people also playing around with this for Conformer. It is always better and cleaner to keep it minimal and not have this hard-coded at this place.

If you want to have your own custom `ConformerEncoderLayer`, or we have some `ConformerEncoderLayerV2`, it can probably still reuse `ConformerConvolutionV1`, as it becomes more unlikely that any further changes are needed.

It makes it more composable.